jib-core/src/main/java/com/google/cloud/tools/jib/docker/CliDockerClient.java

```diff
@@ -246,11 +247,21 @@ public DockerImageDetails inspect(ImageReference imageReference)
       throws IOException, InterruptedException {
     Process inspectProcess =
         docker("inspect", "-f", "{{json .}}", "--type", "image", imageReference.toString());
-    if (inspectProcess.waitFor() != 0) {
-      throw new IOException(
-          "'docker inspect' command failed with error: " + getStderrOutput(inspectProcess));
+
+    try (InputStreamReader stdout =
+        new InputStreamReader(inspectProcess.getInputStream(), StandardCharsets.UTF_8)) {
+
+      String output = CharStreams.toString(stdout);
```

Comment:
I have a concern. `stdout` is an input stream, and at this point, you may not have reached "EOF" of the stream. That is, the process is still running and writing to the stream, which you will miss onward. And hypothetically, you can run in the same issue again.

Reply:
Hi. According to the documentation and source code(https://github.com/google/guava/blob/master/guava/src/com/google/common/io/CharStreams.java) of CharStreams.toString(), it reads all data from the readable. Besides, this is how other parts of this class handles outputs from the docker process.

But of course please do suggest if you have better ideas in mind. ;)

Reply:
It's a stream, and what the doc says is a different thing.

To be correct, you need to keep consuming the stream while the process is running (i.e., looping), because you never know when the process will start/resume writing data on the other side of the stream. And after you noticed the process is terminated, consume the stream once last time.

However, I'd be happy just to see an improvement. So, what I suggest is to read the stream once more after `process.waitFor()`. It's definitely an improvement.

Reply:
lol. I do get what you mean but allow me to elaborate a bit more and you can correct me if I am wrong. :)

![image](https://github.com/GoogleContainerTools/jib/assets/912443/beef6cbd-73b3-4317-8140-b5fe6c78839d)

![image](https://github.com/GoogleContainerTools/jib/assets/912443/d847f570-bd06-49a4-876b-8da7ae0b500d)

Reference for the behavior of InputStream.read() can be found here:
https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-
![image](https://github.com/GoogleContainerTools/jib/assets/912443/ac2679bf-3d1c-4de6-a595-1659cfe958c9)

Reply:
And what's the rationale for using a Reader?

Reply:
Oops. My bad. It is indeed a Reader in our case. Thanks for the correction. :)
Below is the case for Reader.
![image](https://github.com/GoogleContainerTools/jib/assets/912443/af222b33-e454-4615-a42a-916c5f24e8e8)

As you can see, it will still exhaust the underlying Stream. Reference can be found here. https://docs.oracle.com/javase/8/docs/api/java/io/Reader.html
![image](https://github.com/GoogleContainerTools/jib/assets/912443/e744fd0e-4228-4d19-8c20-8cae3fdf0867)

As for the rationale of using a Reader, I was merely following the style of existing code of the same class so that it is consistent.

Reply:
Cool then. Please just leave the comment "CharStreams.toString() will block until the end of stream is reached." That'll save people questioning the same concern as I did.

I'll let the repo maintainer to review and approve this. Ping them if necessary.

Reply:
Good idea! Added the comment as requested. :)